VocaDbModel/Database/Queries/UserQueries.cs

```diff
@@ -516,6 +517,52 @@ class CachedUserStats {
 
 		}
 
+		public (bool created, int reportId) CreateReport(int userId, UserReportType reportType, string hostname, string notes) {
+
+			PermissionContext.VerifyPermission(PermissionToken.ReportUser);
+
+			if (string.IsNullOrEmpty(notes)) {
+				log.Error("Notes are required");
+				return (false, 0);
+			}
+
+			return repository.HandleTransaction(ctx => {
+				var user = ctx.Load(userId);
+
+				ctx.AuditLogger.SysLog($"reporting {user} as {reportType}");
+
+				if (user.GroupId >= UserGroupId.Moderator) {
+					log.Error("Cannot report user with group " + user.GroupId);
+					return (false, 0);
+				}
+
+				if (user.GroupId <= UserGroupId.Regular && reportType == UserReportType.Spamming) {
+					var activeReportCount = ctx.Query<UserReport>()
+						.Where(ur => ur.Entry.Id == userId && ur.Status == ReportStatus.Open && ur.ReportType == UserReportType.Spamming)
+						.ToArray()
+						.Distinct(ur => ur.User.Id)
+						.Count();
+					if (activeReportCount >= 10) {
+						log.Info("User disabled");
+						user.Active = false;
+						ctx.Update(user);
+					} else if (activeReportCount >= 5) {
+						log.Info("User set to limited");
+						user.GroupId = UserGroupId.Limited;
+						ctx.Update(user);
+					}
+				}

```

Comment:
I think this should be done after an entry report has been created, or you'll need 5+1 active reports to set user to limited and 10+1 to disable user.

Reply:
I was thinking it wasn't that important, 5 or 6 reports. But yeah, makes more sense that way. Moved it. Thank you for reviewing!